raddb/sites-available/coa-relay

```diff
@@ -0,0 +1,289 @@
+		# Acct-Session-Id and write each session (which includes to
+		# owning NAS and session ID) to a detail file.
+		#
+		# Returns a single result in the format:
+		#
+		# NasIpAddress1#AcctSessionId1|NasIPAddress2#AcctSessionId2|...
+		#
+		update control {
+
+			# Example MySQL lookup
+			Tmp-String-0 := "%{sql:SELECT IFNULL(GROUP_CONCAT(CONCAT(nasipaddress,'#',acctsessionid) separator '|'),'') FROM (SELECT * FROM radacct WHERE ('%{User-Name}'='' OR UserName='%{User-Name}') AND ('%{Acct-Session-Id}'='' OR acctsessionid = '%{Acct-Session-Id}') AND AcctStopTime IS NULL) a}"
+
+			# Example PostgreSQL lookup
+			# Tmp-String-0 := "%{sql:SELECT STRING_AGG(CONCAT(nasipaddress,'#',acctsessionid),'|') FROM (SELECT * FROM radacct WHERE ('%{User-Name}'='' OR UserName='%{User-Name}') AND ('%{Acct-Session-Id}'='' OR acctsessionid = '%{Acct-Session-Id}') AND AcctStopTime IS NULL) a}"
+
+		}
+		if (&control:Tmp-String-0 != "" && "%{explode:&control:Tmp-String-0 |}") {
+			foreach &control:Tmp-String-0 {
+				if ( "%{Foreach-Variable-0}" =~ /([^#]*)#(.*)/ ) {
+					update request {
+
+						COA-Packet-DST-IP-Address := "%{1}"
+						COA-Acct-Session-Id := "%{2}"
+
+						# Set the CoA/Disconnect port
+						COA-Packet-DST-Port := 1700
+
+						# NASs don't usually want to
+						# see User-Name, and
+						# SQL-User-Name is left over
+						# from the xlat
+						#
+						User-Name !* ANY
+						SQL-User-Name !* ANY
+
+					}
+					if (COA-Packet-Type != "" && \
+					    COA-Packet-DST-IP-Address && \
+					    COA-Acct-Session-Id != "") {
+
+						detail_coa.accounting
+
+					}
+				}
+			}
+		} else {
+			# No sessions found
+			reject
+		}
+
+	}
+}
+
+#
+#  Detail file reader that processes the queue of CoA/Disconnect requests
+#
+server coa-buffered-reader {
+	listen {
+		#
+		#  See sites-available/buffered-sql for more details on
+		#  all the options available for the detail reader.
+		#
+		type = detail
+		filename = "${radacctdir}/detail_coa"
+		load_factor = 90
+		track = yes
+	}
+
+	accounting {
+
+		switch &COA-Packet-Type {
+
+			case "Disconnect-Request" {
+				update {
+					disconnect:Packet-DST-IP-Address = \
+						&COA-Packet-DST-IP-Address
+					disconnect:Packet-DST-Port = \
+						&COA-Packet-DST-Port
+					disconnect:Acct-Session-Id = \
+						&COA-Acct-Session-Id
+					disconnect:Event-Timestamp = "%l"
+					disconnect:Message-Authenticator = 0x00
+				}
+			}
+
+			case "CoA-Request" {
+				update {
+
+					#  Include given attributes
+					coa: += request:[*]
+
+					coa:Packet-DST-IP-Address = \
+						&COA-Packet-DST-IP-Address
+					coa:Packet-DST-Port = \
+						&COA-Packet-DST-Port
+					coa:Acct-Session-Id = \
+						&COA-Acct-Session-Id
+					coa:Event-Timestamp = "%l"
+					coa:Message-Authenticator = 0x00
+
+				}
+			}
+
+		}
+
+		ok
+	}
+}
+
+
+#  The CoA packet is in the "proxy-request" attribute list.
+#  The CoA reply (if any) is in the "proxy-reply" attribute list.
+#
+server originate-coa {
```

Comment:
Maybe rename this to something else, as there's already an `originate-coa` virtual server.